compiler/src/iree/compiler/Codegen/Common/VectorReductionToGPU.cpp

```diff
@@ -151,12 +274,17 @@ static Value groupReduction(Location loc, OpBuilder &builder, Value input,
       Attribute identityAttr =
           getCombiningKindIdentity(builder, kind, loadVal.getType());
       assert(identityAttr && "Unknown identity value for the reduction");
-      Value identity = builder.create<arith::ConstantOp>(loc, identityAttr);
+      Value identity =
+          getCombiningIdentityValue(loc, builder, kind, loadVal.getType());
       loadVal = builder.create<arith::SelectOp>(loc, useIdentityElement,
                                                 identity, loadVal);
     }
     laneVal = warpReduction(loc, builder, loadVal, kind, warpSize, numWarp);
   }
+  // Handles cases for sub-32bit precision where output is still in vector form.
+  if (laneVal.getType().dyn_cast<VectorType>()) {
```

Comment:
nit: should be `laneVal.getType().isa<VectorType>()`

Reply:
done!